src-java/kilda-pce/src/test/java/org/openkilda/pce/finder/DiversityPathFinderTest.java

```diff
@@ -0,0 +1,370 @@
+
+        List<PathSegment> forwardSegments = new ArrayList<>();
+        forwardSegments.add(PathSegment.builder().pathId(FORWARD_PATH_ID).srcSwitch(switchA).srcPort(10)
+                .destSwitch(switchB).destPort(10).build());
+        forwardSegments.add(PathSegment.builder().pathId(FORWARD_PATH_ID).srcSwitch(switchB).srcPort(10)
+                .destSwitch(switchC).destPort(10).build());
+
+        FlowPath forwardPath = FlowPath.builder()
+                .srcSwitch(switchA)
+                .destSwitch(switchC)
+                .pathId(FORWARD_PATH_ID)
+                .segments(forwardSegments)
+                .build();
+        when(flowPathRepository.findById(FORWARD_PATH_ID)).thenReturn(java.util.Optional.of(forwardPath));
+
+
+        List<PathSegment> reverseSegments = new ArrayList<>();
+        reverseSegments.add(PathSegment.builder().pathId(REVERSE_PATH_ID).srcSwitch(switchC).srcPort(10)
+                .destSwitch(switchB).destPort(10).build());
+        reverseSegments.add(PathSegment.builder().pathId(REVERSE_PATH_ID).srcSwitch(switchB).srcPort(10)
+                .destSwitch(switchA).destPort(10).build());
+
+        FlowPath reversePath = FlowPath.builder()
+                .srcSwitch(switchC)
+                .destSwitch(switchA)
+                .pathId(REVERSE_PATH_ID)
+                .segments(reverseSegments)
+                .build();
+        when(flowPathRepository.findById(REVERSE_PATH_ID)).thenReturn(java.util.Optional.of(reversePath));
+
+        Flow flow = getFlow();
+        flow.setSrcSwitch(switchA);
+        flow.setDestSwitch(switchC);
+        flow.setDiverseGroupId(DIVERSE_GROUP_ID);
+        flow.setForwardPathId(new PathId("forward_path_id"));
+        flow.setReversePathId(new PathId("reverse_path_id"));
+        flow.setMaxLatency(Long.MAX_VALUE);
+        flow.setPathComputationStrategy(pathComputationStrategy);
+
+        when(config.getNetworkStrategy()).thenReturn(BuildStrategy.COST.name());
+        when(islRepository.findActiveByBandwidthAndEncapsulationType(flow.getBandwidth(), flow.getEncapsulationType()))
+                .thenReturn(isls);
+
+        when(flowPathRepository.findPathIdsByFlowDiverseGroupId(DIVERSE_GROUP_ID))
+                .thenReturn(Lists.newArrayList(FORWARD_PATH_ID, REVERSE_PATH_ID));
+
+        // check diversity counter
+        AvailableNetwork availableNetwork = availableNetworkFactory.getAvailableNetwork(flow, Collections.emptyList());
+        assertEquals(2, availableNetwork.getSwitch(switchB.getSwitchId()).getDiversityGroupUseCounter());
+
+        // check found path
+        PathComputer pathComputer = new InMemoryPathComputer(
+                availableNetworkFactory, new BestWeightAndShortestPathFinder(200), config);
+
+        GetPathsResult path = pathComputer.getPath(flow);
+        assertThat(path, is(notNullValue()));
+        assertThat(path.getForward(), is(notNullValue()));
+        assertThat(path.getReverse(), is(notNullValue()));
```

Comment:
maybe we should check switches in the path? Path can contain empty segments for example and test will path

Reply:
Fixed